src/tokens/saga.ts

```diff
@@ -221,7 +221,23 @@ export function* fetchTokenBalancesSaga() {
           .toFixed()
       }
     }
-    yield* put(setTokenBalances(tokens))
+
+    /* We are including the fetchedBalancesByTokenId since some balances might be already fetched
+     * so we avoid fetching them again. This could happen if we turn the FETCH_BALANCES_VIA_BLOCKSCOUT flag on.
```

Comment:
```suggestion
     * so we avoid fetching them again. This could happen if the data source includes more tokens than we support (e.g. Blockscout).
```